plugins/orchestrator/src/components/NewWorkflowViewerPage/NewWorkflowViewerPage.tsx

```diff
@@ -0,0 +1,196 @@
+
+import { orchestratorApiRef } from '../../api';
+import {
+  createWorkflowRouteRef,
+  workflowDefinitionsRouteRef,
+} from '../../routes';
+import { BaseOrchestratorPage } from '../BaseOrchestratorPage/BaseOrchestratorPage';
+
+type FormData = {
+  url: string;
+};
+
+export const NewWorkflowViewerPage = () => {
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+  const errorApi = useApi(errorApiRef);
+  const orchestratorApi = useApi(orchestratorApiRef);
+  const createWorkflowLink = useRouteRef(createWorkflowRouteRef);
+
+  const defaultValues: FormData = {
+    url: workflow_json_sample.url,
+  };
+  const { handleSubmit, register, formState } = useForm<FormData>({
+    defaultValues,
+    mode: 'onChange',
+  });
+
+  const { errors } = formState;
+
+  const navigate = useNavigate();
+  const definitionLink = useRouteRef(workflowDefinitionsRouteRef);
+
+  const handleResult = useCallback(
+    async ({ url }: FormData) => {
+      if (!url) {
+        return;
+      }
+      try {
+        const result = await orchestratorApi.createWorkflowDefinition(url);
+
+        if (!result?.definition.id) {
+          errorApi.post(new Error('error importing workflow'));
+        } else {
+          const workflowFormat = result.uri.endsWith('.json') ? 'json' : 'yaml';
+          navigate(
+            definitionLink({
+              workflowId: result.definition.id,
+              format: workflowFormat,
+            }),
+          );
+        }
+      } catch (e: any) {
+        errorApi.post(new Error(e));
+      }
+    },
+    [orchestratorApi, errorApi, navigate, definitionLink],
+  );
+
+  const newWorkflow = useCallback(
+    (format: WorkflowFormat) => {
+      navigate(
+        createWorkflowLink({
+          format,
+        }),
+      );
+    },
+    [createWorkflowLink, navigate],
+  );
+
+  function asInputRef(renderResult: UseFormRegisterReturn) {
+    const { ref, ...rest } = renderResult;
+    return {
+      inputRef: ref,
+      ...rest,
+    };
+  }
+
+  const contentItems = [
+    <Grid item xs={12} xl={6} key="create-workflow">
+      <InfoCard title="Create" subheader="Start authoring from scratch">
+        <Button
+          color="default"
+          variant="outlined"
+          style={{ marginTop: 8, marginRight: 8 }}
+          onClick={() => newWorkflow('yaml')}
+        >
+          YAML
+        </Button>
+        <Button
+          color="default"
+          variant="outlined"
+          style={{ marginTop: 8, marginRight: 8 }}
+          onClick={() => newWorkflow('json')}
+        >
+          JSON
+        </Button>
+      </InfoCard>
+    </Grid>,
+    <Grid item xs={12} xl={6} key="import-workflow">
+      <InfoCard
+        title="Import"
+        subheader="Import an existing workflow from a URL"
+      >
+        <Grid container spacing={2}>
+          <Grid item xs={12}>
+            <form onSubmit={handleSubmit(handleResult)}>
```

Comment:
I think this does not what is expected. We should pass a handler here, not immediately call a function at the render time.

Reply:
There is a refactoring going on and this component will likely be replaced with new code.